Various minor fixes in docs (#512)
yiisoft/validator: docs/guide/result.md

# Result

> The validation result is an object containing errors' data that occurred during validation.

Reviewer: ```suggestion
The validation result is an object containing data of errors occurred during validation.
```

Reviewer: `that` is optional, but is better to change the order like in the suggestion to avoid confusion with "data occurred during validation".

## Errors

Most of the time telling only the status of validation is not enough. There are multiple methods to get detailed errors 

> list with their data from the result. The difference between them is in the grouping, filtering, and representation of every 

Reviewer: Why add the comma before "and"?

Reviewer: https://www.grammarly.com/blog/comma-before-and/

Reviewer: https://github.com/yiisoft/yii2/blob/master/docs/documentation_style_guide.md#general-style

Does Oxford comma apply to Yii3 too?

Author: Doesn't matter for now but it shouyld be consistend and part of the style guide we'll choose.

It's easy to display and iterate, however, with a bigger amount of attributes and depending on a message, it can be
problematic to understand which attribute an error belongs to.

> ##### Error messages not bound to a specific attribute

Reviewer: Why change the level of the header?

];
```

Note that the result is always a 2-dimensional array with attribute names as keys at the first nesting level. This means
that further nesting of attributes is not supported. Returning to the previous example, when `name` and `email` belong 

> to a `user` attribute, the output will be:

Reviewer 2: >  This means
that further nesting of attributes is not supported.

is this also the case when using dot notation for nested attributes?

Reviewer 2: Reading further, it seems that `getErrorMessagesIndexedByPath()` is doing what is not supported by this function, a hint about using the other method for it would be nice.

];
```

Note that the result is always a 2-dimensional array with attribute names as keys at the first nesting level. This means

> that further nesting of attributes is not supported (but could be achieved by using `getErrorMessagesIndexedByPath()`).

Reviewer: Suggest to add the the link to `### Error messages indexed by path`.

];
```

Note that the result is always a 2-dimensional array with attribute names as keys at the first nesting level. This means
that further nesting of attributes is not supported (but could be achieved

> [by using `getErrorMessagesIndexedByPath()`](#error-messages-indexed-by-path)).

Reviewer: ```suggestion
[getErrorMessagesIndexedByPath()](#error-messages-indexed-by-path)).
```

#### Resolving special characters collision in attribute names

> 

Reviewer: Redundant newline.

> When the attribute name contains a path separator (dot - `.` by default) or `Each` rule shortcut (asterisk -`*`), 

Reviewer: What's the reason of this change: "value path" -> "path"?

Author: Because it is "path separator", not "value path separator". It was also suggested by Chris.

> When the attribute name contains a value path separator (dot - `.` by default) or `Each` rule shortcut (asterisk -`*`), 

Reviewer 2: ```suggestion
When the attribute name contains a path separator (dot - `.` by default) or `Each` rule shortcut (asterisk -`*`), 
```
